metrics_sdk/lib/opentelemetry/sdk/metrics/aggregation/explicit_bucket_histogram.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Metrics
+      module Aggregation
+        # Contains the implementation of the ExplicitBucketHistogram aggregation
+        # https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#explicit-bucket-histogram-aggregation
+        class ExplicitBucketHistogram
+          # The Default Value represents the following buckets:
+          # (-inf, 0], (0, 5.0], (5.0, 10.0], (10.0, 25.0], (25.0, 50.0],
+          # (50.0, 75.0], (75.0, 100.0], (100.0, 250.0], (250.0, 500.0],
+          # (500.0, 1000.0], (1000.0, +inf)
+          DEFAULT_BOUNDARIES = [0, 5, 10, 25, 50, 75, 100, 250, 500, 1000].freeze
+
+          def initialize(
+            boundaries: DEFAULT_BOUNDARIES,
+            record_min_max: true
+          )
+            @data_points = {}
```

Comment:
I'm concerned about mutability of the data points. (Where) do we protect against concurrent calls to `update` and `collect`? `collect` returns an array of mutable `HistogramDataPoint`, which we could reasonably document as immutable for callers, but in the `@aggregation_temporality != :delta` case, we subsequently and concurrently mutate them in `update`.

Reply:
I think duping the datapoints on collection is probably the right approach here. cbb93bbac95c345ee0be564542bb8fc40cef7e17

Do you think it makes sense to only dup **if** it's not using delta temporality (that way we get more of a performance win from using delta).

Reply:
> Do you think it makes sense to only dup **if** it's not using delta temporality (that way we get more of a performance win from using delta).

Yep.